src/components/QueryOutput/QueryOutput.js

```diff
@@ -0,0 +1,1988 @@
+      preferredDisplayType
+    )
+  }
+
+  getDisplayTypeFromInitial = (props) => {
+    const isInitialDisplayTypeValid = isDisplayTypeValid(
+      props.queryResponse,
+      props.initialDisplayType,
+      this.tableData?.length,
+      this.pivotTableData?.length
+    )
+
+    // Set the initial display type based on prop value, response, and supported display types
+    const displayType = isInitialDisplayTypeValid
+      ? props.initialDisplayType
+      : this.getUpdatedDefaultDisplayType(props)
+
+    if (!isInitialDisplayTypeValid) {
+      this.displayTypeInvalidWarning(displayType)
+    }
+
+    return displayType
+  }
+
+  hasError = (response) => {
+    try {
+      const referenceIdNumber = Number(response.data.reference_id.split('.')[2])
+      if (referenceIdNumber >= 200 && referenceIdNumber < 300) {
+        return false
+      }
+    } catch (error) {
+      console.warn('Invalid reference ID provided for error')
+    }
+    return true
+  }
+
+  getColumns = () => {
+    if (this._isMounted) {
+      return this.state.columns
+    }
+
+    return this.queryResponse?.data?.data?.columns
+  }
+
+  supportsPivot = () => {
+    const potentialDisplayTypes = this.getPotentialDisplayTypes()
+    return potentialDisplayTypes?.includes('pivot_table')
+  }
+
+  supportsDatePivot = () => {
+    return this.supportsPivot() && this.tableColumns?.length === 2
+  }
+
+  usePivotDataForChart = () => {
+    return this.supportsPivot() && !this.supportsDatePivot()
+  }
+
+  isTableConfigValid = (tableConfig) => {
+    try {
+      if (
+        !tableConfig ||
+        !tableConfig.numberColumnIndices ||
+        !tableConfig.stringColumnIndices ||
+        Number.isNaN(Number(tableConfig.numberColumnIndex)) ||
+        Number.isNaN(Number(tableConfig.stringColumnIndex))
+      ) {
+        return false
+      }
+
+      if (
+        !Array.isArray(tableConfig.numberColumnIndices) ||
+        !Array.isArray(tableConfig.stringColumnIndices)
+      ) {
+        return false
+      }
+
+      const columns = this.getColumns()
+
+      const areNumberColumnsValid = tableConfig.numberColumnIndices.every(
+        (index) => {
+          return columns[index] && isColumnNumberType(columns[index])
+        }
+      )
+      if (!areNumberColumnsValid) {
+        return false
+      }
+
+      const areStringColumnsValid = tableConfig.stringColumnIndices.every(
+        (index) => {
+          return columns[index] && isColumnStringType(columns[index])
+        }
+      )
+      if (!areStringColumnsValid) {
+        return false
+      }
+
+      return true
+    } catch (error) {
+      return false
+    }
+  }
+
+  updateColumns = (columns) => {
+    if (columns) {
+      const formattedColumns = this.formatColumnsForTable(columns)
+      this.setState({
+        columns: formattedColumns,
+        columnChangeCount: this.state.columnChangeCount + 1,
+      })
+      /* ----------- from merge conflict: uncomment this if there is a bug
```

Comment:
How old is this comment?

If it's been tested we're all clear, we should remove it